### PR TITLE
Morpho leverage extension

### PR DESCRIPTION
**Pull request by ckoopmann**

### CODE REVIEW

File: contracts/adapters/MorphoLeverageStrategyExtension.sol [hunk 1]
```diff
@@ -0,0 +1,1281 @@
...
+     * assets deposited in lending protocols for borrowing.
+     *
+     * For lever, max borrow is calculated as:
+     * (Net borrow limit  - existing borrow balance) / collateral price in borrow asset units
+     *
+     * For delever, max repay is calculated as:
+     * Collateral balance  * (net borrow limit - existing borrow balance) / net borrow limit 
+     *
+     * Net borrow limit is calculated as:
+     * Collateral Balance  * Collateral Price in Borrow units *  morpho LLTV * (1 - unutilized leverage %)
```
Comment (MerlinEgalite) [inline contracts/adapters/MorphoLeverageStrategyExtension.sol:1107]: I don't understand why the borrow limit increases when there is more capacity to borrow ie the unutilized leverage percentage increases. This sounds counter intuitive to me.

**Review by Jean-Grimal — APPROVED** [review]

File: test/integration/ethereum/morphoLeverageStrategyExtension.spec.ts [hunk 2]
```diff
@@ -0,0 +1,5238 @@
...
+          const intializeContracts = async () => {
+            await initializeRootScopeContracts();
+
+            await wsteth.approve(debtIssuanceModule.address, ether(1000));
+            // await usdc.approve(debtIssuanceModule.address, ether(1000));
+
+            // Issue 1 SetToken
+            issueQuantity = ether(1);
+
+            await morphoLeverageModule.sync(setToken.address, { gasLimit: 10000000 });
```
Comment (edkim) [inline test/integration/ethereum/morphoLeverageStrategyExtension.spec.ts:741]: I'm curious why you have to call `sync` here? I thought there would be no deposit or borrow positions in Morpho before calling engage.

**Review by edkim — APPROVED** [review]
Looks great to me. I wasn't able to spot any issues. Excellent work 🔥

File: contracts/adapters/MorphoLeverageStrategyExtension.sol (hunk 1 above)
Comment (MerlinEgalite) [reply contracts/adapters/MorphoLeverageStrategyExtension.sol:1117]: Clear to me thx

**Pull request merged**